static/js/activity.js

```diff
@@ -348,9 +354,17 @@ exports.build_user_sidebar = function () {
         return;
     }
 
-    var user_ids = filter_and_sort(presence.get_user_ids());
+    var user_ids;
+
+    if (meta.$user_list_filter.val().length) {
+        user_ids = filter_and_sort(presence.get_user_ids());
+    } else {
+        user_ids = filter_and_sort(people.get_all_persons().map(function (person) {
+            return person.user_id;
+        }));
+    }
 
-    var user_info = _.map(user_ids, info_for);
+    var user_info = _.map(user_ids, info_for).filter(function (person) { return person; });
```

Comment:
What's the story with the extra `.filter()` here?  I don't see what it's needed.

Reply:
Some of them are undefined, so this filters out things that are undefined. This includes bots and yourself.

Reply:
Does it filter undefined?  I'm not convinced.

Reply:
Undefined is a falsey value, and filter only returns elements that return true.

Reply:
Ahh, sure.  Well, anyway, I think if we want to write code like this, it definitely deserves a comment explaining why `undefined` is possible.  But probably we want to remove things that don't have info earlier anyway.

Reply:
I would suggest using `return person !== undefined` though, in case person returns a different falsey value that we want to include.

Reply:
Just look into the `info_for` section to see this in effect:

```js
    // if the user is you or a bot, do not show in presence data.
    if (person.is_bot || person.user_id === page_params.user_id) {
        console.log("totes triggered", person);
        return;
    }
```

<img width="760" alt="screen shot 2017-10-04 at 4 48 25 pm" src="https://user-images.githubusercontent.com/10321399/31205063-ef586358-a923-11e7-8c21-e131209bca07.png">

Reply:
Yeah, follow Cynthia's suggest and make a comment, then.

Nonetheless, this should be using `all_realm_people`.

Reply:
yep it's updated with a comment.

Reply:
We definitely wouldn't want to include anything that is a falsey value that isn't undefined either, since this goes directly into a handlebars method that is searching for properties within an object.

That said, it would be overly defensive programming to check for other falsey values since I don't believe that's possible in this code path.